tests/JMS/Serializer/Tests/Serializer/XmlSerializationTest.php

```diff
@@ -361,6 +363,20 @@ public function testWithoutFormatedOutputByXmlSerializationVisitor()
         $this->assertXmlStringEqualsXmlString($this->getContent('simple_class_object_minified'), $stringXml);
     }
 
+    public function testDiscriminatorAsXmlAttribute()
+    {
+        $xml = simplexml_load_string($this->serialize(new ObjectWithXmlAttributeDiscriminatorChild()));
+        $this->assertEquals('type="child"', trim($xml->xpath('//result/@type')[0]->saveXML()));
```

Comment:
This is an interesting topic about testing, and how much we want to do "unit" testing...

Your test currently tests only of the type is rendered as attribute... but what if for some other change, make the type to be rendered using both strategies.. I mean as attribute and as element? This test will not detect it.

Of course, testing the whole xml content, has some drawbacks, as example will give some "false" failures if we decide to change indentation or other format specific differences... that generally should not make fail your testcase.

To be honest, I prefer to have a false failure, instead of having a not detected failure.

(this means less "unit" testing, and going more in direction of functional or integration tests... (event if the border between this definitions is really blurry)

What is your opinion on it?